android-file-chooser/src/main/java/ir/sohreco/androidfilechooser/FileChooserDialog.java

```diff
@@ -309,7 +308,7 @@ public void onClick(View view) {
         int id = view.getId();
         if (id == R.id.previous_dir_imagebutton) {
             File parent = new File(currentDirectoryPath).getParentFile();
-            if (parent != null && parent.canRead()) {
+            if (parent != null) {
```

Comment:
I don't agree with you. In my project I need to go to root directory. My changes allow me to do that. In your variant my maximum of directory hierarchy is the "storage/emulated/0"